bank_component.rb

```diff
@@ -0,0 +1,97 @@
+=begin
+ Feature 1: 
+ This feature will allow user to either sign up for an account.
+ Or log in with their last name,last 4 digits of their social security number, and password.        
+=end
+
+
+
+#class for log in and sign up method
+class LogInSignUp
+
+    @@user_info=[
+        {first:'andy', last:'park', password:'hello123', ssn:'123456789'},
+        {first:'Jason', last:'kim', password:'bye123', ssn:'135791098'}
+    ]
+
+    #Instance variable to determine if the user's credential is validated
+    @user_credential_validation = false
+    
+    attr_accessor :last_name, :password, :ssn
+
+    def initialize(last_name, password, ssn)
+        @last_name = last_name
+        @password=password
+        @ssn=ssn
+    end
+
+    def validate_log_in_credentials
+        @@user_info.each do |x|
+            if x[:last].to_s == @last_name && x[:password] == @password && x[:ssn][5,8] == @ssn
```

Comment:
I assume this is to grab the last 4 digits. 
Another way to do this would be: 
```
x[:ssn][-4..-1]
```
This at least guarantees that the last 4 digits are always retrieved.

Another option to consider is using Regexes, but unless you're wanting to validate the format, it's probably unnecessary.